lib/bundler/cli/pristine.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+require "bundler/cli/common"
+
+module Bundler
+  class CLI::Pristine
+    def run
+      ::Bundler.load.specs.each do |spec|
+        gem_name = "#{spec.name} (#{spec.version}#{spec.git_version})"
+
+        if spec.source.instance_of? Source::Path
```

Comment:
please change this to switch over `spec.source`, first matching `Rubygems`, then `Git`, then an `else` that prints the warning